Apollo/Components/FadeThumb.cs

```diff
@@ -47,44 +67,87 @@ public class FadeThumb: UserControl {
 
         bool dragged = false;
 
-        void DragStarted(object sender, VectorEventArgs e) {
+        void DragStarted(object sender, VectorEventArgs e)
+        {
             ((Window)this.GetVisualRoot()).Focus();
             dragged = false;
         }
 
-        void DragCompleted(object sender, VectorEventArgs e) {
+        void DragCompleted(object sender, VectorEventArgs e)
+        {
             if (!dragged) Focused?.Invoke(this);
             else if (change != 0) Moved?.Invoke(this, 0, change);
         }
 
-        void MouseDown(object sender, PointerPressedEventArgs e) {
+        void MouseDown(object sender, PointerPressedEventArgs e)
+        {
             PointerUpdateKind MouseButton = e.GetCurrentPoint(this).Properties.PointerUpdateKind;
 
             if (MouseButton != PointerUpdateKind.LeftButtonPressed) e.Handled = true;
-            
+
             ((Window)this.GetVisualRoot()).Focus();
         }
 
-        void MouseUp(object sender, PointerReleasedEventArgs e) {
+        void MouseUp(object sender, PointerReleasedEventArgs e)
+        {
             PointerUpdateKind MouseButton = e.GetCurrentPoint(this).Properties.PointerUpdateKind;
 
-            if (MouseButton == PointerUpdateKind.RightButtonReleased) {
-                Deleted?.Invoke(this);
+            if (MouseButton == PointerUpdateKind.RightButtonReleased)
+            {
+                MenuItem selectedItem = null;
+                foreach (Control item in ThumbContextMenu.Items)
+                {
+                    if (item.GetType() != typeof(MenuItem)) break;
+                    MenuItem menuItem = (MenuItem)item;
+                    menuItem.Icon = "";
+                    if (menuItem.Header.ToString() == FadeType.ToString())
+                    {
+                        selectedItem = menuItem;
+                    }
+                }
+                selectedItem.Icon = this.Resources["SelectedIcon"];
+                MenuOpened?.Invoke(this);
                 e.Handled = true;
             }
         }
 
         double change;
 
-        void MouseMove(object sender, VectorEventArgs e) {
+        void MouseMove(object sender, VectorEventArgs e)
+        {
             if (!dragged) change = 0;
             change += e.Vector.X;
 
             dragged = true;
             Moved?.Invoke(this, e.Vector.X, null);
         }
-        
+
         public void Select() => this.Resources["Outline"] = new SolidColorBrush(new Color(255, 255, 255, 255));
         public void Unselect() => this.Resources["Outline"] = new SolidColorBrush(new Color(0, 255, 255, 255));
+
+        public void OpenMenu()
+        {
+            ThumbContextMenu.Open(Base);
+        }
+        public void ContextMenu_Click(object sender, EventArgs e)
+        {
+            IInteractive item = ((RoutedEventArgs)e).Source;
+
+            if (item.GetType() == typeof(MenuItem))
```

Comment:
`if (item is MenuItem selectedItem)` i probably have more of these that i've written myself that i should look for and fix up